drivers/ethernet/eth_mcux.c

```diff
@@ -107,11 +107,19 @@ struct eth_context {
 
 static void eth_0_config_func(void);
 
+#ifndef CONFIG_NOCACHE_MEMORY
```

Comment:
I think there is no need for #ifdef there. __nocache should always be used as it has no performance impact when DTCM is used.